test/unittest/rtps/persistence/PersistenceTests.cpp

```diff
@@ -28,16 +29,158 @@ class PersistenceTest : public ::testing::Test
+
+        sqlite3_bind_text(add_last_seq_statement, 1, persist_guid, -1, SQLITE_STATIC);
+        sqlite3_bind_int64(add_last_seq_statement, 2, seq_num);
+
+        if (sqlite3_step(add_last_seq_statement) != SQLITE_DONE)
+        {
+            FAIL() << sqlite3_errmsg(db);
+        }
+        sqlite3_finalize(add_last_seq_statement);
+
+        sqlite3_stmt* add_data_statement;
+        sqlite3_prepare_v2(db,"INSERT INTO writers_histories VALUES(?,?,?,?);",-1,&add_data_statement,NULL);
+
+        sqlite3_bind_text(add_data_statement, 1, persist_guid, -1, SQLITE_STATIC);
+        sqlite3_bind_int64(add_data_statement, 2, seq_num);
+        sqlite3_bind_zeroblob(add_data_statement, 3, 16);
+        sqlite3_bind_zeroblob(add_data_statement, 4, 128);
+
+        if (sqlite3_step(add_data_statement) != SQLITE_DONE)
+        {
+            FAIL() << sqlite3_errmsg(db);
+        }
+        sqlite3_finalize(add_data_statement);
+    }
+
+    void add_writer_data(sqlite3* db, int version, const char* persist_guid, int seq_num)
+    {
+        switch (version)
+        {
+            case 0:
+            case 1:
+                add_writer_data_v1(db, persist_guid, seq_num);
+                break;
+            case 2:
+                add_writer_data_v2(db, persist_guid, seq_num);
+                break;
+            default:
+                FAIL() << "unsuppoerted database version " << version;
+        }
+    }
+
+    const char* dbfile = "text.db";
+    const char* create_statement_v1 = R"(
+CREATE TABLE IF NOT EXISTS writers(
+    guid text,
+    seq_num integer,
+    instance binary(16),
+    payload blob,
+    PRIMARY KEY(guid, seq_num DESC)
+) WITHOUT ROWID;
+
+CREATE TABLE IF NOT EXISTS readers(
+    guid text,
+    writer_guid_prefix binary(12),
+    writer_guid_entity binary(4),
+    seq_num integer,
+    PRIMARY KEY(guid, writer_guid_prefix, writer_guid_entity)
+) WITHOUT ROWID;
+)";
+    const char* create_statement_v2 = R"(
+PRAGMA user_version = 2;
+PRAGMA foreign_keys = ON;
+
+CREATE TABLE IF NOT EXISTS writers_states(
+    guid TEXT PRIMARY KEY, -- CHECK(guid REGEXP '([0-9a-fA-F]{1,2}\.){11}[0-9a-fA-F]{1,2}\|([0-9a-fA-F]{1,2}\.){3}[0-9a-fA-F]{1,2}'),
+    last_seq_num INTEGER CHECK(last_seq_num > 0)
+) WITHOUT ROWID;
+
+CREATE TABLE IF NOT EXISTS writers_histories(
+    guid TEXT, -- CHECK(guid REGEXP '([0-9a-fA-F]{1,2}\.){11}[0-9a-fA-F]{1,2}\|([0-9a-fA-F]{1,2}\.){3}[0-9a-fA-F]{1,2}'),
+    seq_num INTEGER CHECK(seq_num > 0),
+    instance BLOB CHECK(length(instance)=16),
+    payload BLOB,
+    PRIMARY KEY(guid, seq_num DESC), 
+    FOREIGN KEY (guid)
+        REFERENCES writers_states(guid)
+) WITHOUT ROWID;
+
+CREATE TABLE IF NOT EXISTS readers(
+    guid text,
+    writer_guid_prefix binary(12),
+    writer_guid_entity binary(4),
+    seq_num integer,
+    PRIMARY KEY(guid, writer_guid_prefix, writer_guid_entity)
+) WITHOUT ROWID;
+)";
```

Comment:
Reuse the statement from Fast-DDS to ease the maintenance of this test. Probably exposing somehow the create method directly.